src/snowflake/snowpark/dataframe.py

```diff
@@ -1125,6 +1126,11 @@ def select(
         exprs = parse_positional_args_to_list(*cols)
         if not exprs:
             raise ValueError("The input of select() cannot be empty")
+        
+        stmt = self._session._ast_batch.assign()
+        ast = stmt.expr
+        ast.sp_dataframe_select__columns.df.sp_dataframe_ref.id.bitfield1 = self._ast_id
+        ast.sp_dataframe_select__columns.variadic = False
```

Comment:
`variadic` is true for some forms – the logic seems to be in `parse_positional_args_to_list`. We have to capture this detail to correctly represent the unparsed form in query history.